src/terminal/TerminalMain.cpp

```diff
@@ -103,8 +103,18 @@ int main(int argc, char** argv) {
         exit(1);
       }
       auto tokens = split(stdinData, '_');
-      idpasskey = tokens[0];
-      FATAL_FAIL(setenv("TERM", tokens[1].c_str(), 1));
+      if (tokens.size() == 2) {
+        idpasskey = tokens[0];
+        FATAL_FAIL(setenv("TERM", tokens[1].c_str(), 1));
+      } else if (tokens.size() == 1) {
+        // Generate a new idpasskey for this session
+        string passkey = genRandomAlphaNum(32);
+        string id = genRandomAlphaNum(16);
+        idpasskey = id + string("/") + passkey;
+        FATAL_FAIL(setenv("TERM", tokens[0].c_str(), 1));
+      } else {
+        LOG(FATAL) << "Invalid number of tokens: " << tokens.size();
+      }
```

Comment:
Btw there're two more `cout` with passkey in this file that we need to remove. (links were in workchat ;)